workspaces/azure-devops/plugins/scaffolder-backend-module-azure-devops/src/actions/devopsRunPipeline.ts

```diff
@@ -116,22 +125,25 @@ export function createAzureDevopsRunPipelineAction(options: {
         project,
         branch,
         templateParameters,
+        pollingInterval,
+        pipelineTimeout,
+        token,
       } = ctx.input;
 
       const url = `https://${host}/${organization}`;
       const credentialProvider =
         DefaultAzureDevOpsCredentialsProvider.fromIntegrations(integrations);
       const credentials = await credentialProvider.getCredentials({ url: url });
 
-      if (credentials === undefined && ctx.input.token === undefined) {
+      if (credentials === undefined && token === undefined) {
         throw new InputError(
           `No credentials provided ${url}, please check your integrations config`,
         );
       }
 
       const authHandler =
-        ctx.input.token || credentials?.type === 'pat'
-          ? getPersonalAccessTokenHandler(ctx.input.token ?? credentials!.token)
+        token || credentials?.type === 'pat'
+          ? getPersonalAccessTokenHandler(credentials!.token)
```

Comment:
I remember I deliberately left the check logic in the function call for a reason, but I can't remember what. 

I'll test it in my set up, I use an AD service Principal to handle auth.

Reply:
Can we revert this change. It's not specifically related to the changes you want to add. I'll also note the way it was matches the the other azure actions: 

https://github.com/backstage/backstage/blob/dce2bfad8c11c9db988b03a5ad666e1cd7f09990/plugins/scaffolder-backend-module-azure/src/actions/azure.ts#L169-L172

I also worry about this breaking for other adopters without a clear reason for this change.